contributing.md

```diff
@@ -0,0 +1,141 @@
+* **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as Markdown code blocks.
+* **Describe the current behavior** and **explain which behavior you expected to see instead** and why.
+* **Explain why this enhancement would be useful** to most users.
+* **List some other applications where this enhancement exists.**
+* **Specify the name and version of the browser you're using.**
+
+### Pull Requests
+
+Please follow these steps to have your contribution considered by the maintainers:
+
+After you submit your pull request, verify that all [Github Actions](https://docs.github.com/en/actions) are passing.
+<details><summary>What if the actions are failing?</summary>If a action is failing, and you believe that the failure is unrelated to your change, please leave a comment on the pull request explaining why you believe the failure is unrelated. If we conclude that the failure was a false positive, then we will open an issue to track that problem with our actions</details>
+
+The reviewer(s) may ask you to complete additional design work, tests, or other changes before your pull request can be ultimately accepted.
+
+## Conventions
+
+### Git
+Git commits should be as granular as possible. When working on a fix for issue X, we try not to add other things we notice (typos, formatting, refactorings, etc.) to the same commit. We don’t use the GitHub shortcuts for closing issues from commits (like fixes #111), since in our process the issue is not solved by the commit but by the reviewed change after it’s deployed to production.
+
+### Commit Messages
+
+* Use the present tense ("Add feature" not "Added feature")
+* Use the imperative mood ("Move cursor to..." not "Moves cursor to...")
+* Use short lines (max 72 chars), and either just one line, or one line, a blank line, and one or more paragraphs.
+* Reference issues after the first line ("See #23")
+
+### Force Pushing
+As a general rule, we never change public commit history, i.e. we don’t use ```--force``` or ```-f``` with ```git push```. The only exception are branches prefixed with ```wip-``` (work in progress), which are considered private to the developer. Pull requests should always be opened for non-wip branches. Local amending and rebasing before pushing to GitHub is no problem and will not require to ```--force``` when pushing. While we consider this general rule as directive, we condone force pushing as long as the branch has no open pull request yet and only one person is working on this branch. In case of a force push we use ```--force-with-lease``` to ensure that we do not overwrite any remote commits. In case of an open pull request, instead of force pushing we open a new branch from master and ```cherry-pick``` needed commits or add new code in this branch. The old branch and the existing pull request should be closed then.
```

Comment:
ah, yes this is what I asked today of in the standup, because it contraicts what is said in https://github.com/metafacture/metafacture-core/wiki/Maintainer-Guidelines#merging-pull-requests , if I understand the "rebase" there correctly, because every rebase forces rewriting history, yes? I think I like thsi guide here better and what rather change the guidelines in metafacture-core. Let's discusss later.

Reply:
Right, also we didn't actually follow that process for Metafacture, we usually merge via GitHub in the PR.